src/layers/XRLayer/XRLayer.js

```diff
@@ -10,13 +10,19 @@ import fs1 from './xr-layer-fragment.webgl1.glsl';
 import fs2 from './xr-layer-fragment.webgl2.glsl';
 import vs1 from './xr-layer-vertex.webgl1.glsl';
 import vs2 from './xr-layer-vertex.webgl2.glsl';
+import { lens, channels } from './shader-modules';
 import { DTYPE_VALUES } from '../../constants';
 import { padColorsAndSliders } from '../utils';
 
+function getSamplerType(props, noWebGL2) {
```

Comment:
Seems like we could clean up all the conditionals in `getShaders` and just return an object from this function with `fs`, `defines`, 'vs':

```suggestion
const SHADER_MODULES = [
	{ fs: fs1, fscmap: fsColormap1, vs: vs1, sampler: 'sampler2D' },
	{ fs: fs2, fscmap: fsColormap2, vs: vs2, sampler: 'usampler2D'},
]
function getShaderProps({ colormap, dtype }, gl) {
  const mod = !isWebGL2(gl) ? SHADER_MODULES[0] : SHADER_MODULES[1];
  if (dtype === '<f4') {
    // force for float32 textures, regardless of GL
  	mod.sampler = 'sampler2D';
  }
  return { 
  	fs: colormap ? mod.fscmap : mod.fs,
  	vs: mod.vs,
  	defines: { 
  	  SAMPLER_TYPE: mod.sampler,
  	  COLORMAP_FUNCTION: colormap || 'viridis'
  	},
  	modules: [project32, picking, channels, lens],
  }
}
```